contracts/CodexTitleMetadata.sol

```diff
@@ -48,22 +49,61 @@ contract CodexTitleMetadata is ERC721Token {
     return imageHashes[_index];
   }
 
-  function modifyDescriptionHash(uint256 _tokenId, bytes32 _newDescriptionHash) public canModify(_tokenId) {
-    tokenData[_tokenId].descriptionHash = _newDescriptionHash;
+  /**
+   * @dev Updates token metadata hashes to whatever is passed in
+   * @param _providerId (optional) An ID that identifies which provider is
+   *  minting this token
+   * @param _providerMetadataId (optional) An arbitrary provider-defined ID that
+   *  identifies the metadata record stored by the provider
+   */
+  function modifyMetadataHashes(
+    uint256 _tokenId,
+    bytes32 _newNameHash,
+    bytes32 _newDescriptionHash,
+    bytes32[] _newImageHashes,
+    string _providerId, // TODO: convert to bytes32?
+    string _providerMetadataId // TODO: convert to bytes32?
+  )
+    public canModify(_tokenId)
+  {
 
-    emit Modified(msg.sender, _tokenId, ModifiedType.DESCRIPTION_CHANGE);
-  }
+    require(exists(_tokenId));
 
-  function modifyNameHash(uint256 _tokenId, bytes32 _newNameHash) public canModify(_tokenId) {
-    tokenData[_tokenId].nameHash = _newNameHash;
+    // nameHash is only overridden if it's not a blank string, since name is a
+    //  required value
+    //
+    // @dev: is this the best way to check for an empty bytes32 array?
+    //  would (_newNameHash != "") be better in any way?
+    //  see: https://ethereum.stackexchange.com/questions/27227/why-does-require-length-of-bytes32-0-not-work
+    if (_newNameHash[0] != 0) {
```

Comment:
```solidity
if (_newNameHash[0] != 0)
```

and

```solidity
if (_newNameHash != "")
```

both seem to work, I'm just not sure of the pros & cons of each. I'm guessing the former is best since it's only comparing one byte instead of all 32, so I went with that. I wonder if there's any case were the first byte could be 0 without the entire array being 0 🤔